src/platform/h563xx/adc_ll.c

```diff
@@ -497,13 +497,32 @@ void ADC_LL_start(ADC_Num adc_num)
         instance->adc_handle, ADC_FLAG_OVR
     ); // Clear any previous flags
 
-    if (HAL_ADC_Start_DMA(
-            instance->adc_handle,
-            (uint32_t *)instance->adc_buffer_data,
-            instance->adc_buffer_size
-        ) != HAL_OK) {
-        THROW(ERROR_HARDWARE_FAULT);
-    } // Start ADC in DMA mode
+    if (adc_num >= ADC_COUNT) {
+        THROW(ERROR_INVALID_ARGUMENT);
+        return;
+    }
+
+    if (g_adc_multimode) {
+        if (HAL_ADCEx_MultiModeStart_DMA(
+                instance->adc_handle,
+                (uint32_t *)instance->adc_buffer_data,
+                instance->adc_buffer_size
+            ) != HAL_OK) {
+            THROW(ERROR_HARDWARE_FAULT);
+        } // Start ADC in DMA mode for multimode
+        return;
+    }
+    if (!g_adc_multimode) {
```

Comment:
isn't this `else`?

Reply:
Since there is a `return` in the previous clause, neither `if` nor `else` are needed here.